django/db/models/aggregates.py

```diff
@@ -78,6 +78,14 @@ def __init__(self, expression, **extra):
         output_field = extra.pop('output_field', FloatField())
         super(Avg, self).__init__(expression, output_field=output_field, **extra)
 
+    def as_oracle(self, compiler, connection):
+        if self.output_field.get_internal_type() == 'DurationField':
```

Comment:
Could this be moved into the Oracle backend, as a class that inherits the general `Avg` and is only registered for `DurationField`? That would seem much cleaner.